R/Rules-class.R

```diff
@@ -1331,71 +1331,64 @@ StoppingTargetProb <- function(target,
   )
 }
 
-# nolint start
-
-## --------------------------------------------------
-## Stopping based on MTD distribution
-## --------------------------------------------------
+# StoppingMTDdistribution ----
 
-##' Stop based on MTD distribution
-##'
-##' Has 90% probability above a threshold of 50% of the current
-##' MTD been reached? This class is used for this question.
-##'
-##' @slot target the target toxicity probability (e.g. 0.33) defining the MTD
-##' @slot thresh the threshold relative to the MTD (e.g. 0.5)
-##' @slot prob required probability (e.g. 0.9)
-##'
-##' @example examples/Rules-class-StoppingMTDdistribution.R
-##' @keywords classes
-##' @export
-.StoppingMTDdistribution <-
-    setClass(Class="StoppingMTDdistribution",
-             representation(target="numeric",
-                            thresh="numeric",
-                            prob="numeric"),
-             prototype(target=0.33,
-                       thresh=0.5,
-                       prob=0.9),
-             contains="Stopping",
-             validity=
-                 function(object){
-                     o <- Validate()
-
-                     o$check(is.probability(object@target,
-                                            bounds=FALSE),
-                             "target must be probability > 0 and < 1")
-                     o$check(is.probability(object@thresh,
-                                            bounds=FALSE),
-                             "thresh must be probability > 0 and < 1")
-                     o$check(is.probability(object@prob,
-                                            bounds=FALSE),
-                             "prob must be probability > 0 and < 1")
+## class ----
 
-                     o$result()
-                 })
-validObject(.StoppingMTDdistribution())
+#' `StoppingMTDdistribution`
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' [`StoppingMTDdistribution`] is the class for stopping based on MTD distribution.
+#' It is used for the cases where the stopping occurs when there exists
+#' `prob` probability above a threshold of `tresh` of the current MTD been
+#' reached. Here, the MTD is defined as the dose for which `prob(DLT) = target`.
```

Comment:
```suggestion
#' reached. Here, the MTD is defined as the dose for which `prob(DLT)` is closest to the target.
```
because we don't get equality to any number?

Reply:
hm can you please check the logic used in https://github.com/Roche/crmPack/blob/main/R/Rules-methods.R and then reword this accordingly? I think the original wording was already misleading

Reply:
> because we don't get equality to any number?

Thanks @danielinteractive, hmm, I think this is the equality (excluding any differences resulting from finite arithmetic precisions issues). See the extract from the corresponding method function:
```
mtdSamples <- dose(x = stopping@target, model, samples)
 ```

Reply:
but I guess `dose()` is doing this via closest one and not via equality right? Otherwise we would need to have infinitely fine dose grid

Reply:
> but I guess `dose()` is doing this via closest one and not via equality right? Otherwise we would need to have infinitely fine dose grid

I think the `dose()` computes the dose regardless the dose grid. See the example below.
```
setMethod(
  f = "dose",
  signature = signature(
    x = "numeric",
    model = "LogisticNormal",
    samples = "Samples"
  ),
  definition = function(x, model, samples) {
    assert_subset(c("alpha0", "alpha1"), names(samples@data))
    alpha0 <- samples@data$alpha0
    alpha1 <- samples@data$alpha1
    ref_dose <- as.numeric(model@ref_dose)
    assert_probabilities(x, len = h_null_if_scalar(alpha0))

    exp((logit(x) - alpha0) / alpha1) * ref_dose
  }
)
```

Reply:
hm ok maybe I have to read this again in detail. somehow the class description is not clear to me right now (also before)

Reply:
Thanks @danielinteractive, I agree, I have just done some changes now, do you think it is ok now?